src/main/java/slimeknights/tconstruct/smeltery/network/ChannelConnectionPacket.java

```diff
@@ -0,0 +1,53 @@
+package slimeknights.tconstruct.smeltery.network;
+
+import io.netty.buffer.ByteBuf;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.network.NetHandlerPlayClient;
+import net.minecraft.network.NetHandlerPlayServer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import slimeknights.mantle.network.AbstractPacketThreadsafe;
+import slimeknights.tconstruct.smeltery.tileentity.TileChannel;
+
+public class ChannelConnectionPacket extends AbstractPacketThreadsafe {
+  protected BlockPos pos;
+  protected EnumFacing side;
+  protected boolean connect;
+
+  public ChannelConnectionPacket() {}
+
+  public ChannelConnectionPacket(BlockPos pos, EnumFacing side, boolean connect) {
+    this.pos = pos;
+    this.side = side;
+    this.connect = connect;
+  }
+
+  @Override
+  public void handleClientSafe(NetHandlerPlayClient netHandler) {
+    TileEntity te = Minecraft.getMinecraft().world.getTileEntity(pos);
+    if(te instanceof TileChannel) {
+      ((TileChannel) te).updateConnection(side, connect);
+    }
+  }
+
+  @Override
+  public void handleServerSafe(NetHandlerPlayServer netHandler) {
+    // clientside only
+    throw new UnsupportedOperationException("Serverside only");
```

Comment:
Should not this be `"Clientside only"`?

Reply:
That's based on the fluid update packet. I assume boni intended it to mean serverside only to send it, but it could have been a typo there